src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
+     *     itinerary start time and the trip has already been checked within the last 60 minutes
+     * - the current time is after the lead time before the next itinerary starts and between 60-15 minutes prior to the
+     *     itinerary start time, but a check has occurred within the last 15 minutes
+     *
+     * These checks are done based off of the information in the trip's journey state's latest itinerary. If no such
+     * itinerary exists or a previous monitored trip's itinerary has completed, then the next possible itinerary will be
+     * calculated and updated in the monitored trip's journey state.
      */
-    public static boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
-        ZoneId zoneId;
-        Optional<ZoneId> fromZoneId = getZoneIdForCoordinates(trip.from.lat, trip.from.lon);
-        if (fromZoneId.isEmpty()) {
-            String message = String.format(
-                "Could not find coordinate's (lat=%.6f, lon=%.6f) timezone for monitored trip %s",
-                trip.from.lat,
-                trip.from.lon,
-                trip.id
-            );
-            throw new RuntimeException(message);
+    public boolean shouldSkipMonitoredTripCheck() throws Exception {
+        // before anything else, return true if the trip is inactive
+        if (trip.isInactive()) return true;
+
+        // calculate the appropriate timezone to use for the target time based off of the appropriate trip end location
+        ZoneId targetZoneId = trip.timezoneForTargetLocation();
+
+        // get the most recent journey state and itinerary to see when the next monitored trip is supposed to occur
+        journeyState = trip.retrieveJourneyState();
+        if (
+            journeyState.matchingItinerary == null ||
+                journeyState.matchingItinerary.endTime.before(new Date(journeyState.lastCheckedMillis))
+        ) {
+            // Either the monitored trip hasn't ever checked on the next itinerary, or the most recent itinerary has
+            // completed and the next possible one needs to be fetched in order to determine the scheduled start time of
+            // the itinerary on the next possible day the monitored trip happens
+
+            LOG.info("Calculating next itinerary for trip");
+            // calculate target time for the next trip plan request
+            // find the next possible day the trip is active by initializing the the appropriate target time. Start by
+            // checking today's date at the earliest in case the user has paused trip monitoring for a while
+            targetZonedDateTime = DateTimeUtils.nowAsZonedDateTime(targetZoneId)
+                .withHour(trip.tripTimeHour())
+                .withMinute(trip.tripTimeMinute())
+                .withSecond(0);
+
+            // if a previous journeyState target date exists, check if the previous target date was today's date. If so,
+            // immediately advance to the next day
```

Comment:
This comment kind of just describes what's happening without any context. A better one might be: "If journeyState indicates that today has already been checked, advance targetDate to tomorrow to find matching itinerary for the next series of checks.".

Also, what does it mean if targetDate is null?

Reply:
Revised comments in 6fd00a976c063c5b46e7254a995fbbb628b90445